packages/hoppscotch-sh-admin/src/helpers/userManagement.ts

```diff
@@ -0,0 +1,121 @@
+import { useToast } from '~/composables/toast';
+import { getI18n } from '~/modules/i18n';
+import { UserDeletionResult } from './backend/graphql';
+import { ADMIN_CANNOT_BE_DELETED, USER_IS_OWNER } from './errors';
+
+type IndividualActionInput = {
+  type: 'individual';
+  metadata: null;
+};
+type BulkActionInput = {
+  type: 'bulk';
+  metadata: {
+    areMultipleUsersSelected: boolean;
+    deletedIDs: string[];
+  };
+};
+
+type IndividualActionResult = {
+  data: null;
+};
+type BulkActionResult = {
+  data: { timeoutID: NodeJS.Timeout | null };
+};
+
+type HandleUserDeletion = {
+  (
+    deletedUsersList: UserDeletionResult[],
+    action: IndividualActionInput | BulkActionInput
+  ): IndividualActionResult | BulkActionResult;
+};
+
+const t = getI18n();
+const toast = useToast();
+
+export const handleUserDeletion: HandleUserDeletion = (
+  deletedUsersList,
+  action
+) => {
+  let timeoutID: NodeJS.Timeout | null = null;
+
+  const uniqueErrorMessages = new Set(
+    deletedUsersList.map(({ errorMessage }) => errorMessage).filter(Boolean)
+  ) as Set<string>;
+
+  const { type, metadata } = action;
+
+  // Show the success toast based on the action type if there are no errors
+  if (uniqueErrorMessages.size === 0) {
+    if (type === 'bulk') {
+      toast.success(
+        metadata.areMultipleUsersSelected
+          ? t('state.delete_user_success')
+          : t('state.delete_users_success')
+      );
+
+      return { type, data: { timeoutID } };
+    }
+
+    toast.success(t('state.delete_user_success'));
+    return { type, data: null };
+  }
+
+  const errMsgMap = {
+    [ADMIN_CANNOT_BE_DELETED]:
+      type === 'bulk'
+        ? t('state.remove_admin_for_deletion')
+        : t('state.remove_admin_to_delete_user'),
+
+    [USER_IS_OWNER]:
+      type === 'bulk'
+        ? t('state.remove_owner_for_deletion')
+        : t('state.remove_owner_to_delete_user'),
+  };
+  const errMsgMapKeys = Object.keys(errMsgMap);
+
+  if (type === 'bulk') {
+    const { areMultipleUsersSelected, deletedIDs } = metadata;
+
+    // Show toast messages with the count of users deleted only if multiple users are selected
+    if (areMultipleUsersSelected) {
+      toast.success(
+        t('state.delete_some_users_success', { count: deletedIDs.length })
+      );
+      toast.error(
+        t('state.delete_some_users_failure', {
+          count: deletedUsersList.length - deletedIDs.length,
+        })
+      );
+    }
+  }
+
+  uniqueErrorMessages.forEach((errorMessage) => {
+    if (errMsgMapKeys.includes(errorMessage)) {
+      if (type === 'bulk') {
+        timeoutID = setTimeout(
```

Comment:
what is the goal here ? show toasts one after the other ?
if that is the case, we could do something like below, so we don't have to deal with the timeouts ourself, and also will help in unifiying the logic between bulk/individual.

```typescript
const messages = [
  'message 1',
  'message 2',
  'message 3',
  'message 4',
  'message 5',
];

const displayMultipleMessages = (messages,currentIndex) => {
  const message = messages[currentIndex];

  toast.show(message, {
    duration: 2000,
    onComplete: () => {
      if (currentIndex < messages.length - 1) {
        displayMultipleMessages(messages, currentIndex + 1);
      }
    },
  });
};
```

Reply:
Yup, that's the goal; made the change in https://github.com/hoppscotch/hoppscotch/pull/3937/commits/63ac2c3cd6b9fb6db9ff36edc73c4d4a0d55cd90.